### PR TITLE
add Person, Student, and Teacher classes.

### PR DESCRIPTION
**Pull request by ismailakinkunmi**
- add Person, Student, and Teacher classes
-  and other requirements

### CODE REVIEW

File: teacher.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(specialization, age, name)
+    @specialization = specialization
+    super(age, name)
```
Comment (sja-thedude) [inline teacher.rb]: - [ ] Kindly make sure to add parent_permission for the super.

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(specialization, age, name)
```
Comment (sja-thedude) [inline teacher.rb]: - [ ] Kindly make sure to initialize parent_permission with the given data in the teacher class, also make sure to add `unknown` value to name.
Example:
`(name = 'unknown', parent_permission: true)`

File: student.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+require './person'
+
+class Student < Person
+  attr_accessor :classroom
+
+  def initialize(classroom, age, name)
```
Comment (sja-thedude) [inline student.rb]: - [ ] Kindly make sure to initialize parent_permission with the given data in the student class, also make sure to add `unknown` value to name.
Example:
`(name = 'unknown', parent_permission: true)`

File: student.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    super(age, name)
```
Comment (sja-thedude) [inline student.rb]: - [ ] Kindly make sure to add parent_permission for the super.

File: person.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
+class Person
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def initialize(age, name = 'unknown', parent_permission: true)
+    @id = id
```
Comment (sja-thedude) [inline person.rb]: - [ ] Kindly give id value like `@id = Random.rand(1..1000)` as per the requirements.

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @ismailakinkunmi, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ Professional Readme
✔️ Good PR title and description
✔️ Linters passing

## Required 
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 5 above)
Comment (ismailakinkunmi) [reply person.rb]: @sja-thedude thanks for the feedback.

**Review by Kingstalux — APPROVED** [review]
Hi @ismailakinkunmi,
# Status: Approved ✔️ ✔️ ✔️ 
128382030-38e7de91-f6f2-45cf-b54e-8068e0446e4c


Good job implementing the required changes 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**